### PR TITLE
Use opamconfig to configure conf-gmp and zarith

### PR DESCRIPTION
**Pull request by michipili**
This is to solve long-standing issues #1464 and #3000.

The technique used and the tool `opamconfig` implementing it have been discussed in #6271 and are also documented in [the opamconfig project itself](https://github.com/michipili/opamconfig).

CC @timbertson,  @djs55 and @avsm as you recently touched these files, please add anyone you think could also be interested in this patch. Thomas Braibant (maintainer of zarith) has been emailed.

### CODE REVIEW

**Comment by michipili** [comment]
Unluckily the zarith package does not seem to have a home anymore, since @gildor478 announced the obsolescence of the historical OCaml forge. This makes it impossible to supply meaningful values to the fields missing from the `opam` file.  @xavierleroy what are your plans to host further development of the `zarith` library?

**Comment by xavierleroy** [comment]
Zarith is now here: https://github.com/ocaml/Zarith

**Comment by michipili** [comment]
I see that I have to add Debian support to `opamconfig`. :)

File: packages/conf-gmp/conf-gmp.1/opam [hunk 1]
```diff
@@ -6,7 +6,10 @@ bug-reports: "https://github.com/ocaml/opam-repository/issues"
 dev-repo: "https://github.com/ocaml/opam-repository.git"
 license: "GPL"
 build: [
-  ["sh" "-exc" "cc -c $CFLAGS -I/usr/local/include test.c"]
+  ["opamconfig" "-d" "gmp:" "-f" "./conf-gmp.config" "sh" "-exc" "cc -c ${CFLAGS} test.c"]
+]
+depends: [
+  "opamconfig" {build & >= "0.3.0"}
```
Comment (avsm) [inline packages/conf-gmp/conf-gmp.1/opam]: It looks like the Alpine musl support has regressed -- the other distros all pass CI.

```
+ opamconfig "-d" "gmp:" "-f" "./conf-gmp.config" "sh" "-exc" "cc -c ${CFLAGS} test.c" (CWD=/home/opam/.opam/4.03.0/build/conf-gmp.1)
[ERROR] The compilation of conf-gmp failed at "opamconfig -d gmp: -f
        ./conf-gmp.config sh -exc cc -c ${CFLAGS} test.c".
```

**Comment by avsm** [comment]
@michipili ping -- just checking if you have any idea how to fix the Alpine version of this patch so it can be merged?

**Comment by avsm** [comment]
@michipili -- just checking in to see if you've had a chance to look at Alpine support. This can't be merged without it since it's a regression for something that already compiles on that platform.

**Comment by avsm** [comment]
No feedback for > 1 month so I'll close this. Please do feel free to reopen if there's a fix for Alpine as its a very useful patch to have.

**Pull request closed without merging**